subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/internal/toolchains/DefaultToolChainSelector.java

```diff
@@ -47,7 +48,11 @@ public DefaultToolChainSelector(ModelRegistry modelRegistry) {
         if (CppPlatform.class.isAssignableFrom(platformType)) {
             t = platformType.cast(new DefaultCppPlatform("current"));
         } else if (SwiftPlatform.class.isAssignableFrom(platformType)) {
-            t = platformType.cast(new DefaultSwiftPlatform("current"));
+            SwiftVersion swiftVersion = SwiftVersion.UNKNOWN;
+            if (toolProvider.isAvailable()) {
+                swiftVersion = SwiftVersion.of(toolProvider.getCompilerMetadata().getVersion());
+            }
+            t = platformType.cast(new DefaultSwiftPlatform("current", swiftVersion));
```

Comment:
If the tool provider is unavailable, then we don't have a Swift platform available, and we should return a `null` result  or use a `Provider` with no value or a `ToolSearchResult` with the reason why the tool is missing, or some other way of signalling that there is no target Swift platform, no tool chain and no tool provider available.

This will need to propagate out to the binary in some way.

Reply:
I pushed off the introduction of `ToolSearchResult` and used an `UnknownSwiftPlatform` implementation instead. If the target platform is `null` is cause some issue downstream. The compiler infrastructure would need some rethinking.